### DISCUSSION TITLE
Question: Support HDFS? And difference with Alluxio?

### DISCUSSION THREAD

**Discussion by zuston** [juicedata/juicefs#2931] [General]

**Comment by davies** [comment]
The HDFS compatible SDK is in the roadmap, currently you can use the Local File System to use JuiceFS in Hadoop. Also, you can use HDFS as an object storage, then you can have POSIX file system by storing the data in HDFS.

Alluxio is a NOT a file system itself, it's a caching layer for other file system or object storage, which means you can not have more feature other than the underlying UFS provide. JuiceFS is a feature complete file system, which only use GET/PUT/DELETE API from object storage.  Both of them can manage the cache for data to speed the underlying object storage up.

**Comment by zuston** [comment]
Thanks.

**Comment by guanbear** [comment]
> The HDFS compatible SDK is in the roadmap, currently you can use the Local File System to use JuiceFS in Hadoop. Also, you can use HDFS as an object storage, then you can have POSIX file system by storing the data in HDFS.
> 
> Alluxio is a NOT a file system itself, it's a caching layer for other file system or object storage, which means you can not have more feature other than the underlying UFS provide. JuiceFS is a feature complete file system, which only use GET/PUT/DELETE API from object storage. Both of them can manage the cache for data to speed the underlying object storage up.

Hi davis @davies ， is it stable enough when use HDFS as an object storage? Is there any user using this in a production environment? We test juicefs + hdfs （as object storage）works well， but I'm not sure if we can use it in production environment .  

Thanks

**Comment by davies** [comment]
We don't have any known issues around this, also we don't know who use HDFS as obect store in production. It's better to test some failure cases to see how it works before putting into production.
